### PR TITLE
fix(dot-repeat): clear dot-repeat after indentation

### PR DESCRIPTION
**Pull request by phgz**
Do not set dot repeat to indentation as it is misleading and messes with undo.

As of now, the default indentation function will perform its action and thus will set the dot-repeat to indent the lines. So when the user types `.` hoping to repeat some surrounds, "nothing" will happen, but it will stack on the undotree. I'm not sure about a workaround for this. I've thought of performing the action in an embedded headless nvim instance and taking back the result and changing the region to the updated text, but it seems overkill.

### CODE REVIEW

**Comment by kylechui** [comment]
Hi there, thanks for the work you've been putting in! I'm not 100% sure what problem this PR aims to address, however. If possible, could you provide a simple before/after behavior change that this is trying to fix?

**Comment by phgz** [comment]
If you perform a surround on new lines, then the indentation function (default one if not changed) will be called. The problem is that calling it will change the dot-repeat behaviour. So If you make a surround, then do `.` several times, nothing will seem to happen. Then, if you try to undo your surround, also it will seems like nothing is happening (this is because we've been just re-applying the indentation function over and over again).

So basically, after this change, the operatorfunc would be set to empty, so trying to repeat the surroundings would *really* do nothing (in fact getting a `E774: 'operatorfunc' is empty`), so undoing will immediately undo your surrounds.

**Comment by kylechui** [comment]
Ah, I see, thanks for the explanation! I just tried it in my own test buffer and was able to replicate it.

**Comment by kylechui** [comment]
Thanks again for all your work!
